### ISSUE TITLE
Redefinition of "__name__" when importing keras callbacks or optimizers

### ISSUE THREAD

**Issue by christian-rauch** [tensorflow/tensorflow#30964]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 1.14.0
- Python version: 2.7.15+
- CUDA/cuDNN version: 10.0

**Describe the current behavior**
When importing all keras callbacks or optimizers via `from tensorflow.keras.callbacks import *` or `from tensorflow.keras.optimizers import *`, the variable `__name__` gets redefined to `tensorflow.keras.callbacks` and `tensorflow.keras.optimizers`.

**Describe the expected behavior**
Variable `__main__` should keep its value `"__main__"` so it can be used to determine if the python code is imported as a module/library or the entry point of a script.

**Code to reproduce the issue**
```Python
from tensorflow.keras.callbacks import *
print(__name__)
```

**Other info / logs**
Only occurs with `tensorflow.keras`, not with the standalone Keras.

**Comment by gadagashwini** [comment]
I am able to reproduce the issue with Tenosrflow 1.14.0 on Colab. Take a look at Colab gist [here](https://colab.research.google.com/drive/1et_k56o8CMt8lWEmwjS6imZKtudiAc1J). Thanks!

**Comment by christian-rauch** [comment]
I don't have access to the gist that you are referring to.
I can reproduce this on a virtual Ubuntu 18.04 machine with Python 2 and 3, and on a CentOS 7 server with Python 2.

**Comment by ymodak** [comment]
This is fixed with tf-nightly version '1.15.0-dev20190731'
```python
from tensorflow.keras.optimizers import *
print(__name__)
__main__
```
